wireapp/wire-android
# feat: Navigate to selected searched message (WPB-4986)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4986" title="WPB-4986" target="_blank"><img alt="Spike" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10820?size=medium" />WPB-4986</a>  [Android] Go to the original message in the conversation 
  </td></table>
  <br />
 








----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When clicking on a searched messages result (a message) it wouldn't navigate to the selected message on Conversation Screen.

### Causes (Optional)

Not implemented.

### Solutions

With Kalium's PR we have the selected message position and are passing it as an offset to [GetMessagesForConversationUseCase.kt](https://github.com/wireapp/wire-android-reloaded/pull/2412/files#diff-6497b06a39a62979866de08efc6714d0c5064b2949dc54a6e88a6f9d26e636ac) so it relies correctly on some messages before and after the selected message.

[ConversationNavArgs.kt](https://github.com/wireapp/wire-android-reloaded/pull/2412/files#diff-146367a9d5b5e35020e5bbc473237c37644778edffac9ebf33f7099157c2d183) now holds a new param: `searchedMessageId` so from `SearchConversationMessagesScreen` we can continue going forward on navigation instead of trying to pass back the value through 2 screens.

There is a new param on [SearchBar.kt](https://github.com/wireapp/wire-android-reloaded/pull/2412/files#diff-5817352408041f0e628777475c6e0019b6a688156af14ebe5104c015bc581182) that is `isLoading` where it was added a CircularProgressIndicator on the left side of the `X (close)` button on the far right, so the user knows that the messages are being fetched and NOT stuck.

### Dependencies (Optional)

Needs releases with:

- [X] [Kalium PR](https://github.com/wireapp/kalium/pull/2186)

### Testing

#### Test Coverage (Optional)


#### How to Test

- Open App
- Open Conversation Screen
- Open Conversation Details Screen
- Open Search Messages Screen
- Search for some messages:

You will notice the following states (please refer to the images added below):
- Default
- No results
- Results
- Loading

### Notes (Optional)

There are still some issues that will be addressed in another time, some of them  are:
- Performance when selecting a very old message and clicking on navigate to bottom (button on the bottom-right side)
- Self deleting messages still appears in search

### Attachments (Optional)


| Default State | No Results State |
| ----------- | ------------ |
|  <img src="https://github.com/wireapp/wire-android-reloaded/assets/5890660/f9aebb10-0eec-42cd-98cb-623648056f10" width="200" height="400" alt="Default" /> |  <img src="https://github.com/wireapp/wire-android-reloaded/assets/5890660/1cb30f9b-85e7-4a01-a0bb-5d70d5647fc9" width="200" height="400" alt="No Results" /> |

| Results State | Loading State |
| ----------- | ------------ |
|  <img src="https://github.com/wireapp/wire-android-reloaded/assets/5890660/10d64efa-06c9-44ba-8e70-5467228c7efc" width="200" height="400" alt="Results" /> |  <img src="https://github.com/wireapp/wire-android-reloaded/assets/5890660/ab252a1f-13bc-41c8-bd55-db2bef86ae8c" width="200" height="400" alt="Loading" /> |
